cmethods/__init__.py

```diff
@@ -1209,218 +395,190 @@ def quantile_delta_mapping(
-        :type value1: float
-        :param value2: Second value to check
-        :type value2: float
-        :return: If any value is NaN or values are equal
-        :rtype: bool
-        """
-        return np.isnan(value1) or np.isnan(value2) or value1 == value2
+        Function to apply a bias correction technique on 1-and 3-dimensional
+        data sets. For more information please refer to the method specific
+        requirements and execution examples.
 
-    @classmethod
-    def ensure_devidable(
-        cls: CMethods,
-        numerator: Union[float, np.ndarray],
-        denominator: Union[float, np.ndarray],
-    ) -> np.ndarray:
-        """
-        Ensures that the arrays can be devided. The numerator will be multiplied by
-        the maximum scaling factor of the CMethods class if division by zero.
-
-        :param numerator: Numerator to use
-        :type numerator: np.ndarray
-        :param denominator: Denominator that can be zero
-        :type denominator: np.ndarray
-        :return: Zero-ensured devision
-        :rtype: np.ndarray | float
+        See https://python-cmethods.readthedocs.io/en/latest/src/methods.html
+
+        :param method: Technique to apply
+        :type method: str
+        :param obs: The reference/observational data set
+        :type obs: XRData
+        :param simh: The modeled data of the control period
+        :type simh: XRData
+        :param simp: The modeled data of the period to adjust
+        :type simp: XRData
+        :param **kwargs: Any other method-specific parameter (like
+            ``n_quantiles`` and ``kind``)
+        :type **kwargs: dict
+        :return: The bias corrected/adjusted data set
+        :rtype: XRData
         """
-        with np.errstate(divide="ignore", invalid="ignore"):
-            result = numerator / denominator
+        kwargs["adjust_called"] = True
+        check_xr_types(obs=obs, simh=simh, simp=simp)
 
-        if isinstance(numerator, np.ndarray):
-            mask_inf = np.isinf(result)
-            result[mask_inf] = numerator[mask_inf] * cls.MAX_SCALING_FACTOR
+        if method == "detrended_quantile_mapping":
+            raise ValueError(
+                "This function is not available for detrended quantile mapping."
+                " Please use cmethods.CMethods.detrended_quantile_mapping"
+            )
 
-            mask_nan = np.isnan(result)
-            result[mask_nan] = 0
-        elif np.isinf(result):
-            result = numerator * cls.MAX_SCALING_FACTOR
-        elif np.isnan(result):
-            result = 0.0
+        # No grouped correction | distribution-based technique
+        if kwargs.get("group", None) is None:
+            return self.__apply_ufunc(method, obs, simh, simp, **kwargs).to_dataset()
 
-        return result
+        if method not in SCALING_METHODS:
+            raise ValueError(
+                "Can't use group for distribution based methods"  # except for DQM
```

Comment:
Why not? You should be able to run e.g. `'time.month'` groupings on QM for example. In this case, you'd just bias correct all the Januaries together and so on. It's pretty common. Many methods suggest a `dayofyear` window of +/- 15 days

Reply:
In my studies I examined different approaches for implementing and applying the bias correction techniques implemented here. After fixing the child diseases, I experimented with grouped / window-based adjustments using distribution-based methods. 

I tried for example applying these techniques on the MPI-ESM1-2-HR and "NOAA-CIRES-DOE Twentieth Century Reanalysis (V3)" for 2 m air temperatures and precipitation (1951-1981-2010). When using QM or QDM using monthly windows, the results have shown to be more similar to the reference data - BUT - when for example calculating the day of year mean for all grid cells - it can be seen that there are anomalies between monthly transitions. Which are the result of monthly+/distribution-based  adjustments, since the distribution for all Januaries are not respected in relation to its surrounding values. I didn't wanted to have this "feature" in this tool since it could lead to confusion. 

If someone is willing to apply groups for non-scaling methods, they could just patch the method - or write a wrapper but I don't want to bring this source of error into this tool. 

I know that a moving window -15 x +15 days for the adjustment of a single date could be a good idea (I've done this in BiasAdjustCXX), but I haven't found time to do more work into this, since I'm not working in research anymore. I think that would be a good feature (+issue) for another release.